[onecta] Initial contribution (#16730)
openhab/openhab-addons: bundles/org.openhab.binding.onecta/README.md

```

## Prerequisites using the binding
Please take the following steps prior to using the binding.
Create a Onecta cloud account in the Onecta app for [Android](https://play.google.com/store/apps/details?id=com.daikineurope.online.controller&hl=en_US) or [iOS](https://apps.apple.com/de/app/onecta/id1474811586?l=en) (if not already done).
Afterwards, pair your Daikin units in the Onecta App.

There is no auto discovery for the Onecta cloud account.
The account is paired using OAuth2 with your Onecta login and the developer credentials obtained from the Onecta Developer Portal.
To pair the account go to the binding's configuration UI at https://<your openHAB address>/onecta. For a standard openHABian Pi installation the address is https://openhabianpi:8443/onecta or https://pi-adres:8443/onecta.
Note that your browser will file a warning that the certificate is self-signed.
This is fine and you can safely continue.
It is NOT possible to use an unsecured connection (http://) for pairing.<br>
Allowed pairing formats are:

> - https://<your-domain>/onecta

Reviewer: ```suggestion

- https://<your-domain>/onecta
```

- https://<your-local-ip>:<your-port>/onecta (standard port 8443)

> - https://home.myopenhab.org/onecta
> - https://openhabianpi:8443/onecta
> 
> **note:** It is **not** allowed to use localhost (Onecta will not accept this)
> - https://localhost:8443
> 
> ## Account Configuration Example
> 
> 
> 
> 
> 

Reviewer: ```suggestion

```

Author: Still working on the readme

Create a Onecta cloud account in the Onecta app for [Android](https://play.google.com/store/apps/details?id=com.daikineurope.online.controller&hl=en_US) or [iOS](https://apps.apple.com/de/app/onecta/id1474811586?l=en) (if not already done).
Afterwards, pair your Daikin units in the Onecta App.

There is no auto discovery for the Onecta cloud account. 
The account is paired using OAuth2 with your Onecta login and the developer credentials obtained from the Onecta Developer Portal. 
To pair the account go to the binding's configuration UI at https://<your openHAB address>/onecta. For a standard openHABian Pi installation the address is https://openhabianpi:8443/onecta or https://pi-adres:8443/onecta. 
Note that your browser will file a warning that the certificate is self-signed. 
This is fine and you can safely continue. 
It is NOT possible to use an unsecured connection (http://) for pairing.<br>
Allowed pairing formats are:
- https://<your-domain>/onecta
- https://<your-local-ip>:<your-port>/onecta (standaard port 8443)
- https://home.myopenhab.org/onecta
- https://openhabianpi:8443/onecta 

It is **not** allowed to use localhost (Onecta will not accept this)

> - https://localhost:8443

Reviewer: ```suggestion
**note:*** It is **not** allowed to use `localhost` (Onecta will not accept this)
```

Once a Onecta account is paired, all supported appliances are automatically discovered as individual things and placed in the inbox. 
They can then be paired with your favorite management UI. 
As an alternative, the binding configuration UI provides a things-file template per paired account that can be used to pair the appliances.

For a detailed walk through the account configuration, see [Account Configuration Example](#account-configuration-example).

## Account Bridge Configuration

| Name             | Type    | Description                                                    | Default | Required | Advanced |
|------------------|---------|----------------------------------------------------------------|---------|----------|----------|

> | `refreshInterval`  | integer | Interval the device is polled in sec.                          | 600     | yes      | no       |

Reviewer: nitpick: please align the table

## Discovered Things Configuration

| Name            | Type    | Description                                                                                              | Default | Required | Advanced |
|-----------------|---------|----------------------------------------------------------------------------------------------------------|---------|----------|----------|
| `unitID`        | text    | UID Unique Identifier. <br>If this thing is created in a thing-file this UID can be found in the logging | N/A     | yes      | no       |
| `refreshDelay`  | integer | Refresh Delay in sec. <br>Only available for thing 'device' and 'domestic-hot-water-tank'                | 15      | yes      | no       |

### Explanation Refresh Delay:
If a command is sent from the binding to OnectaCloud, it needs time to be processed by Daikin. 
This can cause items to flip-flop. <br>
For example: You switch a Unit 'On' with the binding. 
Daikin will process this command and control the unit, this processing can take 15 seconds. 
During this time, the binding may have requested a data refresh from OnectaCloud. 

> If this 'On' command has not yet been processed by Daikin, this will result in the OH item returning to 'Off'. 

Reviewer: ```suggestion
If this 'On' command has not yet been processed by Daikin, this will result in the openHAB item returning to 'Off'. 
```

![Onecta Login Form](doc/onecta-login.png)
When this is the first time you pair an account, you will need to allow openHAB to access your account.
![Onecta Permissions](doc/onecta-permision.png)

When everything worked, you are presented with a page stating that pairing was successful.
Select the locale which should be used to display localized texts in openHAB channels.
From here, you have two options:
Either let the binding automatically configure a bridge instance or copy the presented things-file template to a things-file and return to the overview page.

![Pairing Successful](doc/pairing-success.png)
Once the bridge instance is `ONLINE`, you can either pair things for all appliances via your favorite management UI or use a things-file.
The account overview provides a things-file template that is shown when you expand the account.
This can serve as a starting point for your own things-file.
In the properties of the Onecta account thing, you can also find the available units and their details of your system. 
Sometimes you need to Disable and then re-Enable the account thing to see the updated list of units.

![Account Overview With Bridge](doc/account-overview-with-bridge.png)

> 

Reviewer: ```suggestion
```

Basically all devices connected to Daikin Onecta cloud could be connected with the binding.

> - `acount`: The bidge ensures the connection to Onecta cloud and the recognition of connected units

Reviewer: ```suggestion
- `account`: Acts as bridge and ensures the connection to Onecta cloud and the recognition of connected units
```

- `climate-control`: The unit itself. With this items like climate can controled.
- `gateway`: The network controller of the unit.
- `IndoorUnit`: Gives information about the indoor unit.
- `domestic-hot-water-tank`: Is the thing to control the Hot water tank.

## Discovery

The onecta binding supports auto-discovery of all devices. 
When the bridge is added and connected, it will automatically discover and add onecta devices to the openHAB system. 

> ## Accound / bridge Thing Configuration

Reviewer: ```suggestion
## Account Bridge Configuration
```
